### PR TITLE
JAVA-14870 Update spring-security-kotlin-dsl module under kotlin-tutorials to remove usage of deprecated WebSecurityConfigurerAdapter

### PR DESCRIPTION
**Pull request by anuragkumawat**

### CODE REVIEW

File: spring-security-kotlin-dsl/src/main/kotlin/com/baeldung/security/kotlin/dsl/SpringSecurityKotlinApplication.kt [hunk 1]
```diff
@@ -20,27 +21,31 @@ class SpringSecurityKotlinApplication
 
 @Order(1)
 @Configuration
-class AdminSecurityConfiguration : WebSecurityConfigurerAdapter() {
-    override fun configure(http: HttpSecurity?) {
+class AdminSecurityConfiguration {
+    @Bean
+    open fun filterChainAdmin(http: HttpSecurity): SecurityFilterChain {
```
Comment (kwoyke) [inline spring-security-kotlin-dsl/src/main/kotlin/com/baeldung/security/kotlin/dsl/SpringSecurityKotlinApplication.kt:26]: Does it have to be open?

File: spring-security-kotlin-dsl/src/main/kotlin/com/baeldung/security/kotlin/dsl/SpringSecurityKotlinApplication.kt [hunk 2]
```diff
@@ -20,27 +21,31 @@ class SpringSecurityKotlinApplication
...
+        return http.build()
     }
 }
 
 @Configuration
-class BasicSecurityConfiguration : WebSecurityConfigurerAdapter() {
-    override fun configure(http: HttpSecurity?) {
+class BasicSecurityConfiguration {
+    @Bean
+    open fun filterChainBasic(http: HttpSecurity): SecurityFilterChain {
```
Comment (kwoyke) [inline spring-security-kotlin-dsl/src/main/kotlin/com/baeldung/security/kotlin/dsl/SpringSecurityKotlinApplication.kt:41]: Does it have to be open?

File: spring-security-kotlin-dsl/pom.xml [hunk 3]
```diff
@@ -17,10 +17,12 @@
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-security</artifactId>
+            <version>${spring-boot.version}</version>
```
Comment (kwoyke) [inline spring-security-kotlin-dsl/pom.xml:20]: I think it should be enough to just override the spring-boot.version in the properties section. No need to explicitly set version here. Parent project should handle it correctly. Can you please check it?

File: spring-security-kotlin-dsl/src/main/kotlin/com/baeldung/security/kotlin/dsl/SpringSecurityKotlinApplication.kt (hunk 1 above)
Comment (anuragkumawat) [reply spring-security-kotlin-dsl/src/main/kotlin/com/baeldung/security/kotlin/dsl/SpringSecurityKotlinApplication.kt]: Open keyword is removed

File: spring-security-kotlin-dsl/src/main/kotlin/com/baeldung/security/kotlin/dsl/SpringSecurityKotlinApplication.kt (hunk 2 above)
Comment (anuragkumawat) [reply spring-security-kotlin-dsl/src/main/kotlin/com/baeldung/security/kotlin/dsl/SpringSecurityKotlinApplication.kt]: Open keyword is removed

File: spring-security-kotlin-dsl/pom.xml (hunk 3 above)
Comment (anuragkumawat) [reply spring-security-kotlin-dsl/pom.xml:20]: When I remove explicit declaration, it points to 2.3.3.release 
screenshot attached
2 3 3 release

Comment (kwoyke) [reply spring-security-kotlin-dsl/pom.xml:20]: Don't look at what IntelliJ shows. It's enough to overwrite the property. Once you rebuild the project the correct versions of modules will be used. See printscreen:
image

Comment (anuragkumawat) [reply spring-security-kotlin-dsl/pom.xml]: Changes done
Thanks

**Pull request merged**